dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx

```diff
@@ -0,0 +1,186 @@
+import "./index.less";
+
+import {
+  ActionsColumn,
+  InnerScrollContainer,
+  OuterScrollContainer,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+import { findNoOfDays, formatDateTime } from "utils/dateFunctions";
+import {
+  publishDataset,
+  setSelectedRuns,
+  updateDataset,
+} from "actions/overviewActions";
+import { useDispatch, useSelector } from "react-redux";
+
+import { ProgressBar } from "./common-component";
+import React from "react";
+
+const SavedRunsComponent = () => {
+  const dispatch = useDispatch();
+  const { savedRuns, selectedRuns } = useSelector((state) => state.overview);
+  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+
+  /* Selecting */
+  const areAllRunsSelected =
+    savedRuns?.length > 0 && savedRuns?.length === selectedRuns?.length;
+  const selectAllRuns = (isSelecting) => {
+    dispatch(setSelectedRuns(isSelecting ? savedRuns.map((r) => r) : []));
+  };
+  const onSelectRuns = (run, __rowIndex, isSelecting) => {
+    const otherSelectedRuns = selectedRuns.filter(
+      (r) => r.resource_id !== run.resource_id
+    );
+    const c = isSelecting ? [...otherSelectedRuns, run] : otherSelectedRuns;
+    dispatch(setSelectedRuns(c));
+  };
+  const isRowSelected = (run) =>
+    selectedRuns.filter((item) => item.name === run.name).length > 0;
+  /* Selecting */
+
+  /* Actions Row */
+  const moreActionItems = (dataset) => [
+    {
+      title:
+        dataset.metadata["dataset.access"] === "public"
+          ? "Publish"
+          : "Unpublish",
+      onClick: () => {
+        const accessType =
+          dataset.metadata["dataset.access"] === "public"
+            ? "private"
+            : "public";
+        dispatch(publishDataset(dataset, accessType));
+      },
+    },
+    {
+      title: dataset.metadata["global.dashboard.seen"]
+        ? "Mark unread"
+        : "Mark read",
+      onClick: () =>
+        dispatch(
+          updateDataset(
+            dataset,
+            "read",
+            !dataset.metadata["global.dashboard.seen"]
+          )
+        ),
+    },
+    {
+      title: "Delete",
+      onClick: () => dispatch(deleteDataset(dataset)),
+    },
+  ];
+  /* Actions Row */
+  const makeFavorites = (dataset, isFavoriting = true) => {
+    dispatch(updateDataset(dataset, "favorite", isFavoriting));
+  };
+  const calculatePercent = (dateStr) => {
+    const daysLeft = findNoOfDays(dateStr);
+    return Math.abs(daysLeft / 1000) * 100;
+  };
```

Comment:
Why is the denominator `1000`?  That should presumably be a symbolic constant at least, and it would be better if it were something like the value of `"server. dataset-lifetime"` or the user equivalent of it.

Reply:
I had an interesting idea while chatting with Varshini...

The idea here is that each dataset timeline is being visualized based on an arbitrary 1000 days leading up to the deletion time.

Instead, we could make this a bit more "realistic" by scaling the visualized timeline to the actual lifetime of the dataset. Something like this:

```
dataset.uploaded .. (current date) ............... server.deletion
```

You've already got the `findNoOfDays()`, so basically you're just doing that twice and for "uploaded to current" (part) and "uploaded to deletion" (whole) and then the slider visualizes straightforward % as (whole / part) * 100.

The downside, as Varshini mentioned, is that the total scale of the sliders will no longer be the same down the page, with each slider presumably constrained to the same physical dimensions (I think it'd look awkward otherwise) but representing potentially much different time spans, e.g.,

```
Dataset 1               --------X--------
Dataset 2               --X--------------
Dataset 3               --------------X--
```

Where Dataset 1 might have been uploaded yesterday and set to delete tomorrow (so we're halfway through the lifetime), Dataset 2 was uploaded yesterday and expires in a month, while Dataset 3 was uploaded five years ago and expires next year.

With a common scale factor, that might look more like

```
Dataset 1              -X-
Dataset 2              --X-----------
Dataset 3              ----------------------X------
```

... and that clarifies the scale, but looks awkward, especially if we get the columns compressed to the point where the color bars of the shorter lifetimes become hard to distinguish.

Reply:
I think that noting that we could use `dataset.uploaded` and `server.deletion` to determine the interval is an excellent insight!

However, making all of the bars "consistent" is an over-constrained problem, I think.

If the coloration (or the `X` in Dave's depictions) is supposed to represent the _fraction_ of the lifetime remaining, then we run into the problem of "what's the denominator?" if all the bars are the same length.  We could solve that problem by making the bars have different lengths, but that might be unsightly and potentially confusing.

So, I think we need to decide which attribute to hold fixed...really, what we need to decide is what is most  important for the bars to convey.

Toward that end, I think the most important thing to convey is how much time is left before expiration.  I don't think the life-length is important, except as it affects the user's understanding of how "fast" the bar is moving.

So, I don't think that it is valuable having different bars of different lengths on the screen.  And, as I've been writing this, I think I've convinced myself that, if all the bars are same length on the screen, then it is probably counter-productive to have the bars representing different amounts of time, because that will be misleading about how "fast" the bar is moving.

Thus, I think my recommendation is to pick a fixed interval for the denominator.  I don't know that 1000 days is the right one.  I think the best simple approach would be to use the `"server. dataset-lifetime"` value, because that's a single, fixed number.  It might be better to use the user's maximum lifetime setting, but that might not work well if there are previously-created datasets which have a longer lifetime.  The other option is to make the denominator just the maximum lifetime for all the user's results (or all the results displayed)...but, that might be a little _too_ "Web 2.0".  🥴

Reply:
Yeah, but that's completely arbitrary, and it's hard to visually communicate what the timeframe is. I think an indication of "how far we are through the dataset's lifetime" is an improvement; I think going back to any arbitrary timeframe is losing that.

But maybe this goes back to my first impression when we were shown the use of a progress bar for dataset deletion in that upper left frame: **what the...?** 😕 

Yeah, it's cute to have a visual artifact, but "elevator to what?" comes to mind.

Maybe we'd really be better off dropping that column and color-code the dates. Or maybe even drop the deletion date entirely and simply show how many days, from now, before the dataset will be deleted.

```
Dataset 1               [1 day] (it's on fire 🔥, click 🧯 to put it out)
Dataset 2               30 days (no rush)
Dataset 3               2500 days (yeah, like you care how long that really is)
```

Reply:
I don't think it's _completely_ arbitrary (e.g., it's _way_ less arbitrary than "1000 days"...) -- it is relative to either the server configuration or values set by the user, so there is context for it.

Nevertheless, I'm with you:  the lines are cute, but I don't feel like they are important or even the best way of conveying the information.  That is, we have already made provisions for highlighting soon-to-expire runs in the upper left panel; thus, anything which doesn't appear there we don't need to worry about, and having a bunch of visual clutter for things that we don't need to worry about seems like a waste of attention.

As for displaying a deletion date vs. a days-remaining value, my experience with GitHub has driven home the fact that it's easy to display the wrong one.  (That is, sometimes I want a delta value and sometimes I want an absolute value, and, while I'm not sure I could delineate when I want which, I sure know when I'm presented with the one that isn't what I want.)

My instinct is to remove the line, display the expiration date (as a date), and maybe color it for dates which are "near".

Reply:
Removed the Progress bar :grinning: